### PR TITLE
added new func to compare/get Openshift version

### PR DESCRIPTION
**Pull request by myeung18**
Signed-off-by: Marco Yeung <myeung@redhat.com>

added new func to compare/get Openshift version : pkg openshift
- the funcs use golang.semver for semantic validation, and params are string
OpenShiftVersion struct is now maintained with golang,semver, though we can only retrieve major.minor from the cluster.

### CODE REVIEW

**Comment by bmozaffa** [comment]
@myeung18 what's the status and plan for this?

**Comment by myeung18** [comment]
> @myeung18 what's the status and plan for this?

was planning to come back to this one after finished the ConsoleLink. 
Now I think would get some more work on 3scale: get it properly installed, and confirm if consolelink is to be done by us (should get to know in the next couple of days), then come back to this one.

**Comment by bmozaffa** [comment]
No problem, and no rush. Just wondering.

**Comment by myeung18** [comment]
> No problem, and no rush. Just wondering.

@jeremyary  @bmozaffa    ready for review

File: internal/platform/platform_versioner.go [hunk 1]
```diff
@@ -33,22 +34,24 @@ type K8SBasedPlatformVersioner struct{}
...
+		"1.11+": "v3.11",
+		"1.11":  "v3.11",
+		"1.13+": "v4.1",
+		"1.13":  "v4.1",
+		"1.14+": "v4.2",
+		"1.14":  "v4.2",
+		"1.16+": "v4.3",
+		"1.16":  "v4.3",
+		"1.17+": "v4.4",
+		"1.17":  "v4.4",
```
Comment (bmozaffa) [inline internal/platform/platform_versioner.go:52]: Let's use this opportunity to add 4.5

Comment (myeung18) [reply internal/platform/platform_versioner.go:52]: added.

**Pull request closed without merging**